src/index.js

```diff
@@ -702,17 +653,11 @@ export class ULabel {
             $("#" + this.config["container_id"]).addClass("ulabel-night");
         }
 
-        var images = [document.getElementById(`${this.config["image_id_pfx"]}__0`)];
-        let mappable_images = [];
-        for (let i = 0; i < images.length; i++) {
-            mappable_images.push(images[i]);
-            break;
-        }
-        let image_promises = mappable_images.map(ULabel.load_image_promise);
-        Promise.all(image_promises).then((loaded_imgs) => {
+        let first_bg_img = document.getElementById(`${this.config["image_id_pfx"]}__0`);
+        first_bg_img.decode().then(() => {
             // Store image dimensions
-            that.config["image_height"] = loaded_imgs[0].naturalHeight;
-            that.config["image_width"] = loaded_imgs[0].naturalWidth;
+            that.config["image_height"] = first_bg_img.naturalHeight;
+            that.config["image_width"] = first_bg_img.naturalWidth;
```

Comment:
This looked like a really complicated way to get the first background image. Wrapping it in an array, looping over the array, immediately breaking the loop, mapping a questionable function that directly attaches `.onload` to the element... 

I have made it significantly easier to follow